### PR TITLE
Update to AnnualCountIndexThresholdRecorder to include a range of days to record

### PR DESCRIPTION
**Pull request by adlwk2**
Updates to the recorder add four new arguments to specify start and end dates in a year where an Annual count is to be recorded.

### CODE REVIEW

File: pywr/recorders/_recorders.pyx [hunk 1]
```diff
@@ -1972,6 +1984,18 @@ cdef class AnnualCountIndexThresholdRecorder(Recorder):
 
         if self.exclude_months is not None and ts.month in self.exclude_months:
             return
+        
+        # include a range of dates within a year
+        if self.include_from_month is None or self.include_from_day is None:
+            include_from = 1
```
Comment (jetuk) [inline pywr/recorders/_recorders.pyx:1990]: This should be cdef'd at the top of the method. 

```
cdef int include_from
cdef int include_to
```

File: tests/test_recorders.py [hunk 2]
```diff
@@ -1819,18 +1820,43 @@ def test_annual_count_index_threshold_recorder(
...
+    if exclude_months is None and None in include_range:
         expected_data = [[0, 0], [0, 0], [0, 183], [0, 365], [0, 365]]
+    elif None not in include_range:
+        # Ignore counts between the 1st April and 15th Aug inclusive
+        assert include_range == [4, 1, 8, 15]
+        expected_data = [
+            [0, 0],
+            [0, 0],
+            [0, 30 + 15],
+            [0, 30 + 31 + 30 + 31 + 15],
```
Comment (jetuk) [inline tests/test_recorders.py:1857]: This LGTM.

**Review by jetuk — CHANGES_REQUESTED** [review]
Thanks for the contribution. It looks good. 

I've made some documentation suggestions and I think a couple of the variables should be `cdef`'d.

File: pywr/recorders/_recorders.pyx [hunk 3]
```diff
@@ -1921,13 +1921,25 @@ cdef class AnnualCountIndexThresholdRecorder(Recorder):
     threshold : int
         Threshold to compare parameters against
     exclude_months : list or None
-        Optional list of month numbers to exclude from the count.
+        Optional list of month numbers to include from the count.
```
[the remark this answers is not in the document]
Comment (adlwk2) [reply pywr/recorders/_recorders.pyx:1924]: Nope, I'll change this back

[the remark this answers is not in the document]
Comment (adlwk2) [reply pywr/recorders/_recorders.pyx]: Done

File: pywr/recorders/_recorders.pyx [hunk 4]
```diff
@@ -1921,13 +1921,25 @@ cdef class AnnualCountIndexThresholdRecorder(Recorder):
...
+    include_from_month : int or None
+        Optional start month to specify a range of dates to include from the count.
```
[the remark this answers is not in the document]
Comment (adlwk2) [reply pywr/recorders/_recorders.pyx]: Added to the main docstring and arugment definition

File: pywr/recorders/_recorders.pyx (hunk 1 above)
Comment (adlwk2) [reply pywr/recorders/_recorders.pyx:1996]: Added

**Comment by jetuk** [comment]
Thanks @adlwk2 this looks good!

**Review by jetuk — APPROVED** [review]

**Pull request merged**